src/plugins/expressions/common/expression_types/kibana_datatable.ts

```diff
@@ -23,9 +23,16 @@ import { Datatable, PointSeries } from '.';
 
 const name = 'kibana_datatable';
 
+export interface KibanaDatatableColumnMeta {
+  type: string;
+  indexPatternId: string;
+  params: Record<string, any>;
+}
+
 export interface KibanaDatatableColumn {
   id: string;
   name: string;
+  meta?: KibanaDatatableColumnMeta;
```

Comment:
This is not generic metadata about the table column any more- if it were, it would just include `type`. The addition of `indexPattern` and aggConfigs into the table is not what I would expect for all tables.

Suggestion:

* Move the `type` parameter to the table column
* Create a new table type for `KibanaAggregatedTable`